### DISCUSSION TITLE
Admin Limit Table List with Query?

### DISCUSSION THREAD

**Discussion by yaplej** [filamentphp/filament#2571] [Help] [answered]
Hello,

I am trying to limit a table list to only a sub-set of the records.  I know I can use a filter but I want to actually limit the results displayed based on a field.  How would I go about doing that?  Not seeing anything in the Demo to help with this. 

One thing I tried in my Resource was:
```
    public function getTableRecords(): Collection|Paginator
    {
        return Model::whereNull('record_id');
    }
```

**Comment by danharrin** [comment, marked as answer]
On the list page:

```php
protected function getTableQuery(): Builder
{
    return parent::getTableQuery()->whereNull('record_id');
}
```

https://filamentphp.com/docs/2.x/admin/resources/listing-records#customising-the-eloquent-query

**Reply by yaplej** [reply]
Thank you.
